controller/src/conf/controller.config.properties

```diff
@@ -49,6 +49,7 @@ controller.security.auth.enable=${AUTHORIZATION_ENABLED}
 controller.security.auth.delegationToken.signingKey.basis=${TOKEN_SIGNING_KEY}
 controller.security.auth.delegationToken.ttl.seconds=${ACCESS_TOKEN_TTL_IN_SECONDS}
 controller.security.tls.enable=${TLS_ENABLED}
+controller.security.tls.protocolVersion=${TLS_PROTOCOL_VERSION}
```

Comment:
Given that we're adding a new property to both segment store and controller that indicates the TLS version to be used by each component, this should require a change on pravega-operator to get this value from users. Please open an issue on pravega-operator for the same.

Reply:
@pbelgundi , is it the case?     Aren't these settings fed (blindly) under the PravegaCluster.pravega.options section of the PravegaCluster CRD?

Reply:
@pbelgundi I took the branch for a spin to see if my understanding correct.  It worked as I expected:

I passed these to the PravegaCluster.pravega.options section of the CR spec:
```
controller.security.tls.protocolVersion: TLSv1.3
pravegaservice.security.tls.protocolVersion: TLSv1.3
```

The result was the controllers and segmentstore got these in their respective configmaps:
```
-Dpravegaservice.security.tls.protocolVersion=TLSv1.3
-Dcontroller.security.tls.protocolVersion=TLSv1.3
```

And ultimately responded correctly to my inquieries using openssl s_client -connect, meaning in this case it accepted TLS1.3 and refused 1.2.

So I don't think there are any pravega operator changes required.

Reply:
Thanks for checking this @sarlaccpit . I wasn't sure if this needed changes on operators.

Reply:
Cool; hey btw I had this question too (I'll try it for myself but wondering what you think will happen based on your understanding of the operator):

Say I deploy a PravegaCluster with the options shown above.    I believe those are "opaque" to the operator in the sense that it's not understanding them specifically, it's just passing them along mechanically.

BUT, would the operator react if I patched the PravegaCluster and changed options after the fact?  For example, changing these values from TLSv1.2 to TLSv1.3, on a running cluster let's say.

Even though the operator is not _understanding_ these options, is it _watching_ them (and reacting by respining pods?)?

This is more of a pravega operator question, a tiny bit outside the scope of this PR, but it's related too (meaning if the answer is no, the operator doesn't react, I'll need to create some issue/enhancement)

Reply:
> BUT, would the operator react if I patched the PravegaCluster and changed options after the fact? For example, changing these values from TLSv1.2 to TLSv1.3, on a running cluster let's say.
If I recollect it right, these options are passed as java system properties to the Controller and Segment Store server processes. So after these processes start these properties can be changed by the server. However, incase the process dies and has to be restarted, it would start the processes with the old values (passed to operator) as those are the ones the operator is "aware" of. Hope that explains it :)

Reply:
@pbelgundi yes it explains it.  And what I meant is definitely going through the operator (meaning patching the PravegaCluster CR).   I patched the CR to modify these 2 options and I observed that operator reacted immediately to that, and restarted the SS and CTRL accordingly, with the new settings.